File: 078 Subsets/subsets.cs
public class Solution {	// Author: Xinyu Chen
    public IList<IList<int>> Subsets(int[] nums) {
        Array.Sort(nums);
        int numElements = nums.Length;
        int numSets = 1 << numElements;
        var result = new List<IList<int>>(numSets);
        for (int i = 0; i < numSets; i++)
        {
            var set = new List<int>();
            for (int j = 0; j < numElements; j++)
            {
                if ((i & (1 << j)) != 0)
                {
                    set.Add(nums[j]);
                }
            }
            result.Add(set);
        }
        return result;
    }
}

File: 078 Subsets/subsets.cpp
class Solution {
public:
    vector<vector<int>> subsets(vector<int>& nums) {
        vector<int> entry;
        helper(nums, entry, 0);
        return result;
    }
    
    void helper(vector<int>& nums, vector<int>& entry, int curr) {
        result.push_back(entry);
        for (int i = curr; i < nums.size(); i++) {
            entry.push_back(nums[i]);
            helper(nums, entry, i + 1);
            entry.pop_back();
        }
    }

private:
    vector<vector<int>> result;
}; // Author: XC
